File: DynamicProgramming1/PadovanSequence/CSPadovanSequence/PadovanSequence.cs
using System;
using System.Collections.Generic;

namespace CSPadovanSequence
{
    static class Constants
    {
        public const int Max = 100;
    }

    class PadovanSequence
    {
        static List<long> padovanSequence = new List<long>(Constants.Max + 1);

        static void getPadovanSequence()
        {
            padovanSequence.Add(0);
            padovanSequence.Add(1);
            padovanSequence.Add(1);
            padovanSequence.Add(1);

            for (int i = 4; i <= Constants.Max; i++)
                padovanSequence.Add(padovanSequence[i - 2] + padovanSequence[i - 3]);
        }

        static void Main(string[] args)
        {
            getPadovanSequence();

            int t = Convert.ToInt32(Console.ReadLine());

            for (int i = 0; i < t; i++)
            {
                int n = Convert.ToInt32(Console.ReadLine());
                Console.WriteLine(padovanSequence[n]);
            }
        }
    }
}

File: DynamicProgramming1/PadovanSequence/CPadovanSequence/PadovanSequence.c
#include <stdio.h>

#define MAX 100

long long padovan_sequence[MAX + 1] = {0, 1, 1, 1};

void get_padovan_sequence()
{
	for (int i = 4; i <= MAX; i++)
		padovan_sequence[i] = padovan_sequence[i - 2] + padovan_sequence[i - 3];
}

int main()
{
	int t, n;

	get_padovan_sequence();

	scanf_s("%d", &t);

	for (int i = 0; i < t; i++)
	{
		scanf_s("%d", &n);
		printf("%lld\n", padovan_sequence[n]);
	}

	return 0;
}
